zlib-rs/src/deflate/longest_match.rs

```diff
@@ -190,26 +190,35 @@ fn longest_match_help<const SLOW: bool>(
         // - >= 8: scan_end is bounded by `258 - (4 + 2 + 1)`, so an 8-byte read is in-bounds
         // - >= 4: scan_end is bounded by `258 - (2 + 1)`, so a 4-byte read is in-bounds
         // - >= 2: scan_end is bounded by `258 - 1`, so a 2-byte read is in-bounds
+        let mut len = 0;
         unsafe {
-            if best_len < core::mem::size_of::<u32>() {
+            if best_len < core::mem::size_of::<u64>() {
+                let scan_val = u64::from_ne_bytes(
```

Comment:
I also tried `if best_len < 32 {` followed by the equivalent AVX2 implementation (similar to how `compare32` processes each 32=byte chunk), but it was slower overall.